hitbox-core/src/request.rs

```diff
@@ -20,7 +24,11 @@ pub trait CacheableRequest
 where
     Self: Sized,
 {
-    async fn cache_policy<P, E>(self, predicates: P, extractors: E) -> RequestCachePolicy<Self>
+    async fn cache_policy<P, E>(
+        self,
+        predicates: P,
+        extractors: E,
+    ) -> Result<RequestCachePolicy<Self>, PredicateError>
```

Comment:
I used a specific error here - PredicateError. I don't see any other places where a different error could come from. So far, so good - but it's not generalized. Your thoughts? @singulared